lib/ddtrace/contrib/sinatra/tracer.rb

```diff
@@ -152,6 +152,14 @@ def render(engine, data, *)
               span.set_tag('sinatra.route.path', @datadog_route)
               span.set_tag(Datadog::Ext::HTTP::STATUS_CODE, response.status)
 
+              # Adding tags just because this was supposed to be working already
+              span.set_tag('span.trace_id', request.env['HTTP_X_DDTRACE_PARENT_TRACE_ID'])
+              span.set_tag('span.parent_id', request.env['HTTP_X_DDTRACE_PARENT_SPAN_ID'])
```

Comment:
As I understand your comment, those are empty, that is, `request.env['HTTP_X_DDTRACE_PARENT_TRACE_ID']` has no valid trace id? I would try and debug-puts this and see what's in `request.env` in that case. The documentation refers to a working example with Ruby 2.2.6 and Sinatra 1.4.8, but obviously it lacks details as by reading your code, I can't spot what the exact problem is.